core/src/Telemetry/XMsTelemetryInfo.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System.Globalization;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Identity.Core.Telemetry
+{
+    internal class XMsTelemetryInfo
+    {
+        public string Version { get; set; }
+        public string ServerErrorCode { get; set; }
+        public string ServerSubErrorCode { get; set; }
+        public string TokenAge { get; set; }
+        public string SpeInfo { get; set; }
+    }
+
+    internal static class XMsTelemInfoHelper
+    {
+        internal static XMsTelemetryInfo parseXMsTelemHeader(string headerValue, RequestContext requestContext)
```

Comment:
This is not tested and it should be very easy to test. Just pass in some headerValues and ensure that parsing is ok or that each of the failure conditions is met. You can use a mocked logger to verify the errors.